examples/Example/src/new_api/drag_n_drop/DragAndDrop.tsx

```diff
@@ -0,0 +1,281 @@
+import Draggable from './Draggable';
+
+if (Platform.OS === 'android') {
+  if (UIManager.setLayoutAnimationEnabledExperimental) {
+    UIManager.setLayoutAnimationEnabledExperimental(true);
+  }
+}
+
+export interface DraggableItemData<T> {
+  data: T;
+  isActive: boolean;
+  draggingActive: boolean;
+  tileSize: number;
+}
+
+interface DragAndDropProps<T> {
+  items: T[];
+  itemsInRowCount: number;
+  onOrderUpdate: (items: T[]) => void;
+  renderItem: (data: DraggableItemData<T>) => React.ReactElement;
+  rowGap?: number;
+  columnGap: number;
+}
+
+function DragAndDrop<T extends { id: string }>({
+  items,
+  renderItem,
+  onOrderUpdate,
+  itemsInRowCount,
+  columnGap,
+  rowGap = columnGap,
+}: DragAndDropProps<T>) {
+  const { TILE_SIZE, TILE_WITH_MARGIN_SIZE, ROW_HEIGHT } = getSizeConstants({
+    itemsInRowCount,
+    rowGap,
+    columnGap,
+  });
+
+  const PlaceholderComponent = { id: 'dnd-transparent' } as T;
+  const [data, setData] = useState(items);
+  const [dragActive, setDragActive] = useState(false);
+  const [activeElementId, setActiveElementId] = useState<null | string>(null);
+  const [placeholderIndex, setPlaceholderIndex] = useState<null | number>(null);
+  const activeElementTranslation = {
+    x: useSharedValue(0),
+    y: useSharedValue(0),
+  };
+  // used to calculate position of currently dragged element in relation to drag start position
+  // we are using dragged element center as calculations point - then we just add translationX/Y from gesture
+  // this way we can easily calculate proper index based of current gesture position
+  const [
+    activeElementInitialPosition,
+    setActiveElementInitialPosition,
+  ] = useState({
+    x: 0,
+    y: 0,
+  });
+
+  const updateDataOnDragEnd = () => {
+    const newData = [...data];
+    const activeElementIndex = getActiveElementIndex();
+    if (activeElementIndex !== null && placeholderIndex !== null) {
+      newData.splice(activeElementIndex, 1);
+      const activeElement = getActiveElementById();
+      if (activeElement) {
+        newData.splice(placeholderIndex, 0, activeElement);
+      }
+    }
+    setData(newData);
+    onOrderUpdate(newData);
+  };
+
+  const enableDragging = (id: string) => {
+    setActiveElementId(id);
+    setDragActive(true);
+  };
+
+  const updateDataOnEnd = () => {
+    setDragActive(false);
+    setActiveElementId(null);
+    updateDataOnDragEnd();
+    activeElementTranslation.x.value = 0;
+    activeElementTranslation.y.value = 0;
+  };
+
+  const onDragEnd = () => {
+    const newPlacePosition = getItemCenterPosition(placeholderIndex);
+    if (newPlacePosition !== null) {
+      activeElementTranslation.x.value = withTiming(
+        newPlacePosition.x - activeElementInitialPosition.x,
+        { duration: 100 }
+      );
+      activeElementTranslation.y.value = withTiming(
+        newPlacePosition.y - activeElementInitialPosition.y,
+        { duration: 100 }
+      );
+      setTimeout(updateDataOnEnd, 120);
```

Comment:
Why do we need `setTimeout` here?

Reply:
We want to wait till animation to new  place will finish. I think there are bugs in RN2 with how withTiming callback works so we can't use it :(